lib/deploy/lifecycleHooks.js

```diff
@@ -1,43 +1,78 @@
 'use strict';
 
+var _ = require('lodash');
 var RSVP = require('rsvp');
 var utils = require('../utils');
 var chalk = require('chalk');
-var process = require('child_process');
+var cprocess = require('child_process');
 var FirebaseError = require('../error');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+
+
+function runCommand(command, options) {
+  return new Promise(function(resolve, reject) {
+    var child = cprocess.spawn(command, _.assign({}, options, {
+      shell: true,
+      stdio: [0, 1, 2] // Inherit STDIN, STDOUT, and STDERR
+    })
+    );
+    logger.info('Running command: ' + command);
+    console.log(process.env);
+    child.on('error', function(err) {
+      reject(new Error(err));
+    });
+    child.on('exit', function(code, signal) {
+      if (signal !== null) {
+        reject(new Error('Command terminated with signal ' + signal));
+      } else if (code) {
+        reject(new Error('Command terminated with non-zero exit code' + code));
+      } else {
+        resolve();
+      }
+    });
+  });
+}
 
 module.exports = function(target, hook) {
   // Errors in postdeploy script will not exit the process since it's too late to stop the deploy.
   var exit = hook !== 'postdeploy' ? undefined : {exit: 2};
 
   return function(context, options) {
-    var command  = options.config.get(target + '.' + hook);
-    if (!command) {
+    var commands  = options.config.get(target + '.' + hook);
+
+    if (!commands) {
       return RSVP.resolve();
     }
+    if (typeof commands === 'string') {
+      commands = [commands];
+    }
+    // active project ID
+    process.env.GCLOUD_PROJECT = getProjectId(options);
+    // root directory where firebase.json can be found
+    process.env.PROJECT_DIR = options.projectRoot;
+    // location of hosting site
+    if (target === 'hosting') {
+      process.env.RESOURCE_DIR = options.config.path(options.config.get('hosting').public);
+    }
+    // location of functions deploy
+    if (target === 'functions') {
```

Comment:
I'd change this to a switch statement and as default, set RESOURCE_DIR to match PROJECT_DIR (or possibly the location of whatever resource we're deploying since arguably database/storage rules can also be in a different directory than PROJECT_DIR)